### PR TITLE
fix concurrent test relying on sleep

### PR DESCRIPTION
**Pull request by jbardin**
Make an old concurrent test more deterministic, and not rely on sleep for any synchronization.

### CODE REVIEW

**Review by pselle — COMMENTED** [review]
It took me a while to understand this and I _think_ I do, but come with a meta question -- what is the particular purpose of this test and could we add a comment so one doesn't have to guess? I'm thinking it's parallel creation of resources (using different providers?)

A nit: using `begin` as the name of a CancelFunc was a bit 😕 and perhaps in the realm of "clever" (not preferred for tests) but I think some grounding in the answer to the first question might make me care less about the naming.

**Comment by jbardin** [comment]
Good questions! I'll also add these as comments into the PR, but the test is verifying that the `parallelism` flag is correctly limiting the concurrency within the apply process. There a note in there already about the mock provider locking all operations, which is why it requires multiple instances for concurrency. 

I thought using `begin` was being more descriptive! The context here is being used as a starting gate to align the initial batch of apply calls, to ensure they are concurrent. Since cancelling the context starts the process, I went and used `begin` to indicate that.

File: command/apply_test.go [hunk 1]
```diff
@@ -182,13 +185,15 @@ func TestApply_parallelism(t *testing.T) {
...
-	// This blocks all the apply functions. We close it when we exit so
-	// they end quickly after this test finishes.
-	block := make(chan struct{})
-	// signal how many goroutines have started
-	started := make(chan int, 100)
+	// started is a semaphore that we use to ensure that we never have more
+	// than "par" apply operations happening concurrently
+	started := make(chan struct{}, par)
 
-	runCount := &hwm{}
```
Comment (alisdair) [inline command/apply_test.go:191]: I think the `hwm` struct is now unused and maybe could be removed?

**Review by alisdair — APPROVED** [review]
This makes sense to me, thanks for the additional explanatory comments!

File: command/apply_test.go (hunk 1 above)
Comment (jbardin) [reply command/apply_test.go:191]: Thanks, I meant to do that!

**Review by pselle — APPROVED** [review]
😄 thanks for the comments! and a little 🤦 for me about it being the `-parallelism` flag, I missed that very important detail/forgot about that flag.

**Pull request merged**